usr/share/rear/skel/default/etc/scripts/system-setup

```diff
@@ -222,23 +243,37 @@ if automatic_recovery ; then
     done 2>&1
 fi
 
-# In unattended mode launch "rear --non-interactive recover" automatically 
-# i.e. with automated "reboot" (reboot_command) after successful recovery:
+# In unattended_recovery mode call RECOVERY_COMMANDS automatically
+# plus automated calling REBOOT_COMMANDS after successful recovery:
 if unattended_recovery ; then
     choices=( "View Relax-and-Recover log file(s)"
-              "Go to Relax-and-Recover shell"
+              "Login at the rescue system"
             )
-    echo -e "\nLaunching 'rear recover' automatically in unattended (non-interactive) mode\n"
-    if rear $rear_debug_options --non-interactive recover ; then
-        echo -e "\n'rear recover' finished successfully\n"
-        echo -e "\n'$REBOOT_COMMANDS_LABEL' in $USER_INPUT_INTERRUPT_TIMEOUT seconds (Ctrl-C to interrupt)\n"
+    eval echo "Launching \'$RECOVERY_COMMANDS_LABEL\' automatically"
```

Comment:
Why is it preferable to use `eval echo` instead of just `echo`? I would prefer the latter, I find it much more readable. Every time I see an `eval`, I must think really hard about what gets actually executed and how and where the expansion happens.
Moreover, `eval` of a dynamically constructed string is generally a very suspect construct. See https://xkcd.com/327/.
Try setting `RECOVERY_COMMANDS_LABEL='; rm -rf / ; echo '` and executing the line above.
(I don't expect the user to put somewhere like this in the label, and the input can be considered trusted, so it is not a security issue. But the user can still put there stuff like `"` or `'` or `|` and expect them to appear on the screen, not be interpreted by the shell. Such characters are actually used sometimes for nice display.)

Reply:
In contrast, `eval "$command"` below should be OK, one expects the $command string to be actually evaluated by the shell.

Reply:
Or using the `P` transformation operator:
```
RECOVERY_COMMANDS=( 'echo $rear_options recover' )
rear_options='foo bar'
for command in "${RECOVERY_COMMANDS[@]}" ; do echo Running RECOVERY_COMMANDS ${command@P} ; done
```
This does not work on RHEL 7, which uses bash 4.2.46, though - the feature is too new.

Reply:
> The most suitable option seems to be the `envsubst` utility. But it would introduce a dependency on another package (`gettext`).

This does not work either - as it is not a bash builtin, the variables need to be exported.

Reply:
What about this:
```
envsubst_internal() {
(
  for v in $(envsubst -v "$1"); do
    export $v
  done
  envsubst <<<"$1"
)
}
RECOVERY_COMMANDS=( 'echo $rear_options recover' )
rear_options='foo bar'
for command in "${RECOVERY_COMMANDS[@]}" ; do echo Running RECOVERY_COMMANDS $(envsubst_internal "${command}") ; done
```

Reply:
I think it becomes too complicated
and I don't like to depend on gettext
because gettext seems to much only for that.

I would prefer something simpler.

Currently I think replacing only $rear_options with its value
should be sufficient and the $rear_options value is safe because
we set it in [usr/share/rear/skel/default]/etc/scripts/system-setup

E.g. something like
```
# unset command rear_options something

# command='echo "first" ; $rear_options ; echo "second" ; $rear_options $something'

# rear_options='ls qqq'

# something="/etc/fstab"

# echo "Running '$command' automatically" | sed -e "s/\$rear_options/$rear_options/g"
Running 'echo "first" ; ls qqq ; echo "second" ; ls qqq $something' automatically

# eval $command
first
ls: cannot access 'qqq': No such file or directory
second
ls: cannot access 'qqq': No such file or directory
/etc/fstab
```

Reply:
I believe that gettext (at least, gettext-runtime) will be installed on most systems because GRUB depends on it: the GRUB utilities invoke gettext at many places.

Reply:
With
https://github.com/rear/rear/commit/9b39e433f958a80c5fead1263bad66f90cad0d6b
I do no longer call
```
eval echo "... $command ..."
```
but instead I just call plain
```
echo "... $command ..."
```
(the default `command='rear $rear_options recover'`)
to keep the code simple and fail-safe because
```
Better very simple code than oversophisticated (possibly fragile) constructs.
```
cf.
https://github.com/rear/rear/wiki/Coding-Style

Reply:
Accordingly with
https://github.com/rear/rear/commit/b131e473807f28c051f1c019d17d63c245a406da
I use the simple default
```
RECOVERY_COMMANDS_LABEL='rear recover'
```
instead of
```
RECOVERY_COMMANDS_LABEL='rear $rear_options recover'
```
because the latter would need fragile code like
```
eval echo "... $RECOVERY_COMMANDS_LABEL ..."
```
or complicated code to safely evaluate nested variables
but this is only meant as some meaningful user info
about what is going on and not as debug info what the
exact called command is.

Reply:
@pcahyna 
provided you do not object
I would like to merge the current simple code
tomorrow afternoon.

If more sophisticated user info is really neded
that properly evaluates even nested variable values
things can be improved later via a separated pull request.

Reply:
@jsmeix

We need the code to be first of all reliable. It should be also simple if possible. Your last version has both qualities, so I like it, even if the information shown for user is not perfect.

Reply:
I have one more remark about the change: the previous change had some `echo -e "\n...\n"`. Now you removed the "\n", which will result in less empty lines in the output. Is that intentional? (It may be even preferable, because the previous version could have some unintended results if the $command contains some backslash escapes.)

Reply:
Yes, less empty lines in the output is intended
in particular to avoid that information gets
too fast "out of sight" by scrolling.
So I removed some '\n'.